tests/unit_tests/test_tethys_apps/test_base/test_app_base.py

```diff
@@ -1543,3 +1544,36 @@ def test_remove_from_db_2(self, mock_ta, mock_log):
 
         # Check tethys log error
         mock_log.error.assert_called()
+
+    def test_navigation_links_not_auto(self):
+        app = tethys_app_base.TethysAppBase()
+        app.nav_links = ["test", "1", "2", "3"]
+        links = app.navigation_links
+        self.assertListEqual(links, app.nav_links)
+
+    def test_navigation_links_auto_excluded_page(self):
+        app = tethys_app_base.TethysAppBase()
+        app.nav_links = "auto"
+        app.index = "home"
+        app.root_url = "test-app"
+
+        app._registered_url_maps = [
+            Namespace(name="exclude_page", title="Exclude Page", index=-1),
```

Comment:
I just wanted a mock-like instance of UrlMap... I should have just used a mock.MagicMock object, lol. I'll swap it for that.

Reply:
Turns out I must've chosen `Namespace` for a reason. Using `mock.MagicMock` causes the tests to fail because mocking the property `name` cannot be done on initialization (see [mock docs](https://docs.python.org/3/library/unittest.mock.html#mock-names-and-the-name-attribute)). While I could have broken it out into two steps, it seemed like using `Namespace` shouldn't be a big deal. Let me know if you disagree.

Reply:
Try using the `UrlMaps` for the test instead of the `Namespace`.